test/types/object.js

```diff
@@ -2265,6 +2288,40 @@ describe('object', () => {
             Helper.validate(Joi.compile(schema), [[obj, true, { a: 10, b: 10 }]]);
         });
 
+        it('can rename from regex', () => {
+
+            const schema = Joi.object({
+                a: Joi.number(),
+                b: Joi.number()
+            }).rename(/a/, 'b', { alias: true });
+
+            const obj = { a: 10 };
+            Helper.validate(Joi.compile(schema), [[obj, true, { a: 10, b: 10 }]]);
```

Comment:
Are the calls to `Joi.compile()` doing anything in these tests?  I see the same in the existing test above too, but I find it a bit confusing, as I would expect it to just end-up here: https://github.com/hapijs/validate/blob/master/lib/compile.js#L66-L68